core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1159,6 +1180,25 @@ class UnifiedLog(@volatile var logStartOffset: Long,
       validBytesCount, lastOffsetOfFirstBatch, Collections.emptyList[RecordError], LeaderHwChange.NONE)
   }
 
+  /**
+   * Return true if the record batch has a higher leader epoch than the specified leader epoch
+   *
+   * @param batch the batch to validate
+   * @param origin the reason for appending the record batch
+   * @param leaderEpoch the epoch to compare
+   * @return true if the append reason is replication and the batch's partition leader epoch is
+   *         greater than the leader epoch, otherwise false
```

Comment:
nit: maybe this could be further adjusted to 
`and the batch's partition leader epoch is greater than specified leaderEpoch, otherwise false`

Reply:
Done.